### PR TITLE
sof-tplgreader.py: document boolean precedence and other design issues

### PR DESCRIPTION
**Pull request by marc-hb**
Also remove unused imports and add pydocs about data structures. No
functional change.

Signed-off-by: Marc Herbert <marc.herbert@intel.com>

### CODE REVIEW

**Comment by marc-hb** [comment]
No new pylint warning in https://github.com/thesofproject/sof-test/pull/714/checks?check_run_id=2877298503

**Review by fredoh9 — DISMISSED** [review]
Thank you for adding detailed document and comments!

**Comment by marc-hb** [comment]
Thanks @fredoh9 ! For this one I want to for @aiChaoSONG 's review too, he wrote bits of this code.

**Review by aiChaoSONG — APPROVED** [review]
Thanks

File: tools/sof-tplgreader.py
Comment (marc-hb) [inline tools/sof-tplgreader.py]: I screwed up a small bit of this explanation, just pushed a fixed version now.

**Review by marc-hb — COMMENTED** [review]
Last minute two-lines long `--help` fix, need your approval again. Sorry for the inconvenience.

**Review by aiChaoSONG — APPROVED** [review]

**Comment by marc-hb** [comment]
2 approvers and pure doc update: can be fixed incrementally later if something is found wrong. Merging.

**Pull request merged**
